packages/core/src/baseclient.ts

```diff
@@ -602,29 +602,27 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
       },
     );
   }
+}
 
-  /**
-   * Verifies that return value of configured `beforeSend` is of expected type.
-   */
-  protected _ensureBeforeSendRv(
-    rv: PromiseLike<Event | null> | Event | null,
-  ): PromiseLike<Event | null> | Event | null {
-    const nullErr = '`beforeSend` method has to return `null` or a valid event.';
-    if (isThenable(rv)) {
-      return (rv as PromiseLike<Event | null>).then(
-        event => {
-          if (!(isPlainObject(event) || event === null)) {
-            throw new SentryError(nullErr);
-          }
-          return event;
-        },
-        e => {
-          throw new SentryError(`beforeSend rejected with ${e}`);
-        },
-      );
-    } else if (!(isPlainObject(rv) || rv === null)) {
-      throw new SentryError(nullErr);
-    }
-    return rv;
+/**
+ * Verifies that return value of configured `beforeSend` is of expected type.
+ */
+function _ensureBeforeSendRv(rv: PromiseLike<Event | null> | Event | null): PromiseLike<Event | null> | Event | null {
+  const nullErr = '`beforeSend` method has to return `null` or a valid event.';
+  if (isThenable(rv)) {
+    return (rv as PromiseLike<Event | null>).then(
```

Comment:
The casting here to PromiseLike should be unnecessary with the changes from #4278